### ISSUE TITLE
Patterns: core 'featured' category not registered

### ISSUE THREAD

**Issue by Andrew-Starr** [WordPress/gutenberg#40443]
### Description

As of 13.0 the core `featured` pattern category does not seem to be registered, and the category is missing from the pattern section of the block inserter.

If trying to recreate this issue with Twenty Twenty-Two theme you will need to comment out L17 of `/twentytwentytwo/inc/block-patterns.php` as this theme registers the category itself.

```
'featured' => array( 'label' => __( 'Featured', 'twentytwentytwo' ) ),
```

Prior to Gutenberg 13.0 the core `featured` category was available without the theme needing to register it, so I'm not sure if this is a bug or whether it is intended.

### Step-by-step reproduction instructions

Activate a theme that does not register the `featured` pattern category
_or_ remove/comment out where the theme is registering the `featured` category

Edit a post/page, select the 'Patterns' tab in the block inserter.
Notice the core 'Featured' category is missing from the dropdown selector.

### Environment info

WP 5.9.3
Gutenberg 13.0
Twenty Twenty-Two theme, and other themes

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ndiego** [comment]
I have confirmed this. The "Featured" category did used to exist by default. @Mamaduka do you know if something may have changed with the updates to the Patterns API or perhaps the Patterns Directory integration?

**Comment by Mamaduka** [comment]
@ndiego, it looks like related to the new API changes.

The category is registered before loading remote patterns. Now, this function is only called via REST API, so it's not available during the normal WP lifecycle.

**Comment by ndiego** [comment]
Hmm, I am on the fence about whether we should consider this a "bug", or simply a "change". It will impact themes that are currently using the "Featured" category without registering it themselves and place patterns in the "Uncategorized" category, but perhaps all categories that a theme wants to use should be registered by the theme itself. 🤔 

It's hard to tell if this was an intentional change or an oversight. The easiest solution would be to add it back if possible, but I could be convinced it should not have been registered to begin with.

**Comment by Mamaduka** [comment]
Probably an oversight on our part. The category is also used for featured patterns from Patterns Directory.

**Comment by ndiego** [comment]
This is an issue we really should try and get resolved for 6.0. I know anecdotally, there are many FSE theme authors that are using the default "featured" category. 

cc @annezazu @priethor
